dump.c

```diff
@@ -877,33 +877,36 @@ void
 Perl_dump_sub_perl(pTHX_ const GV *gv, bool justperl)
 {
     CV *cv;
+    PerlIO * debug_log;
+    bool is_gv;
 
     PERL_ARGS_ASSERT_DUMP_SUB_PERL;
 
-    cv = isGV_with_GP(gv) ? GvCV(gv) : CV_FROM_REF((SV*)gv);
+    cv = (is_gv = cBOOL(isGV_with_GP(gv))) ? GvCV(gv) : CV_FROM_REF((SV*)gv);
```

Comment:
I'm not fond of this sort of assignment inside assignment, why not split it up in two statements to make things easier to read?